### ISSUE TITLE
`Safari TP` is unrecognised

### ISSUE THREAD

**Issue by romainmenke** [mrhenry/core-web#1746]
@romainmenke Thank you for your reply.

Yes, I have tried passing `browserslist`. But, I [directly call](https://github.com/AnYiEE/AwesomeGadgets/blob/master/scripts/modules/utils/build-util.ts#L233) the `babel.transformAsync` method instead of `module.exports` as in the example you provided, it seems that `babel-plugin-core-web` cannot obtain the correct browserslist and throws an error as follows:

```
node_modules\.pnpm\@mrhenry+core-web@1.2.0\node_modules\@mrhenry\core-web\lib\index.js:300
    throw new Error(`Unexpected browserslist result ${range}\nplease file an issue at https://github.com/mrhenry/core-web/issues`);
          ^

Error: [BABEL]: Unexpected browserslist result safari TP
please file an issue at https://github.com/mrhenry/core-web/issues (While processing: "D:\\GitRepository\\AwesomeGadgets\\node_modules\\.pnpm\\@mrhenry+babel-plugin-core-web@1.2.0\\node_modules\\@mrhenry\\babel-plugin-core-web\\lib\\index.js")
    at parseRange (node_modules\.pnpm\@mrhenry+core-web@1.2.0\node_modules\@mrhenry\core-web\lib\index.js:300:11)
    at node_modules\.pnpm\@mrhenry+core-web@1.2.0\node_modules\@mrhenry\core-web\lib\index.js:69:61
    at Array.map (<anonymous>)
    at required (node_modules\.pnpm\@mrhenry+core-web@1.2.0\node_modules\@mrhenry\core-web\lib\index.js:69:52)
    at module.exports (node_modules\.pnpm\@mrhenry+babel-plugin-core-web@1.2.0\node_modules\@mrhenry\babel-plugin-core-web\lib\index.js:5:21)
    at async (node_modules\.pnpm\@babel+core@7.23.5\node_modules\@babel\core\lib\gensync-utils\async.js:35:33)
...
```

Then, I directly pass [the browserslist in the project](https://github.com/AnYiEE/AwesomeGadgets/blob/master/.browserslistrc#L4) to it, like this:

```
plugins: [
	["@mrhenry/core-web", {
		browserslist: [
			"fully supports es5 and not dead",
		]
	}]
]
```

I received the same error as above (`Unexpected browserslist result ${range}\nplease file an issue at https://github.com/mrhenry/core-web/issues`), but I tried using the browserslist strings example like yours, and it can parse correctly.

So, I had to passing it the "browsers" option instead, but it seems there are also some issues as certain browser names cannot be matched one-to-one, resulting in the inability to correctly polyfill.

_Originally posted by @AnYiEE in https://github.com/mrhenry/core-web/issues/1745#issuecomment-1847888720_

**Comment by romainmenke** [comment]
@AnYiEE I've split this out into it's own issue because it is something we can trivially resolve.

It seems that browserslist started returning `Safari TP` at some point, which is not a value we expect.

We can filter this out.

**Comment by romainmenke** [comment]
We no longer throw on versions that end with `tp` or `beta`. (case insensitive match)
This patch has been released in `v1.2.1`
